src/Init/Data/Int/DivMod/Lemmas.lean

```diff
@@ -2894,6 +2894,71 @@ theorem bmod_eq_self_of_le_mul_two {x : Int} {y : Nat} (hle : -y ≤ x * 2) (hlt
     x.bmod y = x :=
   bmod_eq_of_le_mul_two hle hlt
 
+/- ### ediv -/
+
+theorem ediv_lt_self_of_pos_of_ne_one {x y : Int} (hx : 0 < x) (hy : y ≠ 1) :
+    x / y < x := by
+  by_cases hy' : 1 < y
+  · rw [Int.div_def]
+    unfold Int.ediv
+    obtain ⟨xn, hx'⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+    obtain ⟨yn, hy'⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
+    simp only [hx', hy', Int.ofNat_eq_coe, Int.ofNat_lt]
+    apply Nat.div_lt_self (by omega) (by omega)
+  · have := @Int.ediv_nonpos_of_nonneg_of_nonpos x y (by omega) (by omega)
+    omega
+
+theorem ediv_nonneg_of_nonneg_of_nonneg {x y : Int} (hx : 0 ≤ x) (hy : 0 ≤ y) :
+    0 ≤ x / y := by
+  rw [Int.div_def, ]
+  unfold Int.ediv
+  obtain ⟨xn, hx'⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+  obtain ⟨yn, hy'⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
+  simp only [hx', hy', Int.ofNat_eq_coe, Int.ofNat_lt]
+  exact Int.ofNat_zero_le (xn / yn)
```

Comment:
Please try to avoid `Int.div_dev` and unfolding `Int.ediv`. This should never be necessary. For example, this proof can just be
```suggestion
  obtain ⟨xn, rfl⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
  obtain ⟨yn, rfl⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
  rw [← Int.ofNat_ediv]
  exact Int.ofNat_zero_le (xn / yn)
```
Similar changes are possible in the lemmas above and below.